aws_lambda_powertools/utilities/data_classes/kinesis_firehose_event.py

```diff
@@ -0,0 +1,113 @@
+import base64
+import json
+from typing import Iterator, Optional
+
+from aws_lambda_powertools.utilities.data_classes.common import DictWrapper
+
+
+class KinesisFirehoseRecordMetadata(DictWrapper):
+    @property
+    def _metadata(self) -> dict:
+        """Optional: metadata associated with this record; present only when Kinesis Stream is source"""
+        return self["kinesisRecordMetadata"]  # could raise KeyError
+
+    @property
+    def shard_id(self) -> Optional[str]:
+        """Kinesis stream shard ID; present only when Kinesis Stream is source"""
+        return self._metadata.get("shardId")
+
+    @property
+    def partition_key(self) -> Optional[str]:
+        """Kinesis stream partition key; present only when Kinesis Stream is source"""
+        return self._metadata.get("partitionKey")
+
+    @property
+    def approximate_arrival_timestamp(self) -> Optional[str]:
```

Comment:
```suggestion
    def approximate_arrival_timestamp(self) -> int:
```

Reference: https://github.com/awslabs/aws-lambda-powertools-python/pull/1556#discussion_r989106935